commit/tokenprice/validate_observation.go

```diff
@@ -0,0 +1,64 @@
+package tokenprice
+
+import (
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set/v2"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+func (p *processor) ValidateObservation(
+	prevOutcome Outcome,
+	query Query,
+	ao plugincommon.AttributedObservation[Observation],
+) error {
+	obs := ao.Observation
+
+	if err := validateFChain(obs.FChain); err != nil {
+		return fmt.Errorf("failed to validate FChain: %w", err)
+	}
+
+	observerSupportedChains, err := p.chainSupport.SupportedChains(ao.OracleID)
+	if err != nil {
+		return fmt.Errorf("failed to get supported chains: %w", err)
+	}
+
+	for chain := range obs.FChain {
+		if !observerSupportedChains.Contains(chain) {
+			return fmt.Errorf("chain %d is not supported by observer", chain)
+		}
+	}
+
+	if err := validateObservedTokenPrices(obs.FeedTokenPrices); err != nil {
+		return fmt.Errorf("failed to validate observed token prices: %w", err)
+	}
+
+	return nil
+}
+
+func validateFChain(fChain map[cciptypes.ChainSelector]int) error {
+	for chainSelector, f := range fChain {
+		if f < 0 {
```

Comment:
I don't think we should allow 0 fChain.
```suggestion
		if f <= 0 {
```

Reply:
same for other validations in other files.